From 93a2ca67dd972426c6681e599a6d5d825f753739 Mon Sep 17 00:00:00 2001
From: Michal Schmidt <mschmidt@redhat.com>
Date: Tue, 17 Feb 2015 10:33:01 +0100
Subject: [PATCH] journal-remote: fix certificate status memory leak

The output of gnutls_certificate_verification_status_print() needs to be
freed.

Noticed this while staring at verify_cert_authorized() to see what could
possibly confuse gcc5 on armv7hl to segfault during compilation.

(cherry picked from commit 9c3cf9693ac5c0a332ba376f99e6adea28b1bb0d)
(cherry picked from commit 7057db8b36594bb03c611cab711cd992ad4de31e)
---
 src/journal-remote/microhttpd-util.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/journal-remote/microhttpd-util.c b/src/journal-remote/microhttpd-util.c
index 55c45f4693..f73ee39020 100644
--- a/src/journal-remote/microhttpd-util.c
+++ b/src/journal-remote/microhttpd-util.c
@@ -184,6 +184,7 @@ static int verify_cert_authorized(gnutls_session_t session) {
         }
 
         log_info("Certificate status: %s", out.data);
+        gnutls_free(out.data);
 
         return status == 0 ? 0 : -EPERM;
 }
